src/UI/Utils/usePrompt.tsx

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useContext, useEffect } from "react";
+import { UNSAFE_NavigationContext as NavigationContext } from "react-router-dom";
```

Comment:
UNSAFE ??

Reply:
I know, but I didn't find other way to halt and resume react-router navigation in v6, and there is no other solution online that doesn't use this

Reply:
normally window.alert does that natively but you're using a different implementation